### PR TITLE
feat: type modifier on imports

### PR DESCRIPTION
**Pull request by snewcomer**
WIP - trying to get this syntax to compile....

Previously, as on Typescript 3.8, we had the ability to import types, albeit on a new line.

```
import A from 'foo';
import type { AType } from 'foo';
```

As of 4.5, we can inline that type, making it easier to manage top level imports.

```
import A, { type AType } from 'foo';
```

fixer PR (in progress): https://github.com/typescript-eslint/typescript-eslint/pull/5050.  Once this is merged, I'll run it on the whole repo.
follow up: https://github.com/emberjs/ember.js/pull/20087

Docs: https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#type-on-import-names

### CODE REVIEW

File: packages/@ember/-internals/routing/lib/system/route.ts [hunk 1]
```diff
@@ -31,25 +30,27 @@ import { dependentKeyCompat } from '@ember/object/compat';
...
 import type { Template, TemplateFactory } from '@glimmer/interfaces';
-import type {
-  InternalRouteInfo,
-  ModelFor,
-  Route as IRoute,
-  Transition,
-  TransitionState,
+import { 
+  PARAMS_SYMBOL, 
+  STATE_SYMBOL
```
Comment (Josh-Cena) [inline packages/@ember/-internals/routing/lib/system/route.ts:35]:
```suggestion
  STATE_SYMBOL,
```

**Comment by kategengler** [comment]
Is this still relevant? It is a draft PR

**Comment by NullVoxPopuli** [comment]
I think it got implemented via https://github.com/emberjs/ember.js/pull/20593 and prior PRs.

**Comment by kategengler** [comment]
Closing as completed

**Pull request closed without merging**